File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/actions/ClawReturn.java
package org.firstinspires.ftc.teamcode.robot.actions;

import androidx.annotation.NonNull;

import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
import com.acmerobotics.roadrunner.Action;

import org.firstinspires.ftc.teamcode.robot.Lift;

public class ClawReturn implements Action {
    private boolean init = false;
    private Lift claw;

    @Override
    public boolean run(@NonNull TelemetryPacket packet) {
        if(!init) {
            claw.pitchHome();
            claw.clawOpen();
            init = true;
        }

        return false;
    }

    public ClawReturn(Lift li) {
        claw = li;
    }

    public Action clawReturn(Lift li) {
        return new org.firstinspires.ftc.teamcode.dairy.actions.ClawReturn(li);
    }
}

File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/dairy/actions/ClawReturn.kt
package org.firstinspires.ftc.teamcode.dairy.actions

import com.acmerobotics.dashboard.telemetry.TelemetryPacket
import com.acmerobotics.roadrunner.Action
import org.firstinspires.ftc.teamcode.dairy.subsystems.OuttakeClaw
import org.firstinspires.ftc.teamcode.dairy.util.Waiter

class ClawReturn() : Action {
    private var init = false
    private var i = 0
    private lateinit var waiter:Waiter

    override fun run(p: TelemetryPacket): Boolean {
        if (!init) {
            waiter = Waiter()
            waiter.start(200)
            OuttakeClaw.pitch_pos = OuttakeClaw.pitch_down
            init = true
        }

        if(waiter.isDone && i == 0) {
            OuttakeClaw.claw_pos = OuttakeClaw.claw_open
            i = 1
        }

        return i == i
    }

    companion object {
    fun clawReturn(): Action {
        return ClawReturn()
    }}
}